### PR TITLE
Adjust kyma components to Kubernetes 1.16

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**
Adjust kyma components to Kubernetes 1.16

Changes proposed in this pull request:
- Adjust kyma components to Kubernetes 1.16

**Related issue(s)**

See also: kyma-project/kyma#5530

### CODE REVIEW

File: resources/core/charts/kubeless/templates/tests/e2e-test.yaml [hunk 1]
```diff
@@ -36,7 +36,8 @@ spec:
       serviceAccount: e2e-{{ .Chart.Name }}-tests
       containers:
       - name: e2e-{{ .Chart.Name }}-tests
-        image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.kubeless_integration_tests.dir }}e2e-kubeless-integration-test:{{ .Values.global.kubeless_integration_tests.version }}
+        # image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.kubeless_integration_tests.dir }}e2e-kubeless-integration-test:{{ .Values.global.kubeless_integration_tests.version }}
+        image: eu.gcr.io/sap-se-cx-kyma-wookiee/e2e-kubeless-integration-test:kubernetes-1.16.3
```
Comment (m00g3n) [inline resources/core/charts/kubeless/templates/tests/e2e-test.yaml]: will be updated in image-bump PR

File: tests/end-to-end/upgrade/chart/upgrade/templates/pod.yaml [hunk 2]
```diff
@@ -41,4 +41,5 @@ spec:
     - name: APP_DEX_USER_SECRET
       value: "{{ .Values.dex.userSecret }}"
     - name: APP_TESTING_ADDONS_URL
-      value: "https://github.com/kyma-project/addons/releases/download/0.8.0/index-testing.yaml"
+      # value: "https://github.com/kyma-project/addons/releases/download/0.8.0/index-testing.yaml"
+      value: "github.com/kyma-project/addons//addons/index-testing.yaml"
```
Comment (m00g3n) [inline tests/end-to-end/upgrade/chart/upgrade/templates/pod.yaml]: will be updated in image-bump PR

Comment (m00g3n) [inline tests/end-to-end/upgrade/chart/upgrade/templates/pod.yaml]: will be updated in image-bump PR

File: tests/end-to-end/upgrade/chart/upgrade/templates/test/test-upgrade.yaml [hunk 3]
```diff
@@ -35,5 +35,6 @@ spec:
         - name: APP_DEX_USER_SECRET
           value: "{{ .Values.dex.userSecret }}"
         - name: APP_TESTING_ADDONS_URL
-          value: "https://github.com/kyma-project/addons/releases/download/0.8.0/index-testing.yaml"
+          # value: "https://github.com/kyma-project/addons/releases/download/0.8.0/index-testing.yaml"
+          value: "github.com/kyma-project/addons//addons/index-testing.yaml"
```
Comment (m00g3n) [inline tests/end-to-end/upgrade/chart/upgrade/templates/test/test-upgrade.yaml]: will be updated in image-bump PR

File: resources/core/charts/kubeless/templates/tests/test.yaml [hunk 4]
```diff
@@ -19,7 +19,8 @@ spec:
       serviceAccount: {{ .Chart.Name }}-tests
       containers:
       - name: tests
-        image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.kubeless_tests.dir }}kubeless-tests:{{ .Values.global.kubeless_tests.version }}
+        # image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.kubeless_tests.dir }}kubeless-tests:{{ .Values.global.kubeless_tests.version }}
+        image: eu.gcr.io/sap-se-cx-kyma-wookiee/kubeless-tests:kubernetes-1.16.3
```
Comment (m00g3n) [inline resources/core/charts/kubeless/templates/tests/test.yaml]: will be updated in image-bump PR

File: resources/monitoring/templates/tests/test.yaml [hunk 5]
```diff
@@ -20,7 +20,8 @@ spec:
       serviceAccount: {{ .Chart.Name }}-tests
       containers:
       - name: tests
-        image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.monitoring_integration_tests.dir }}{{ .Values.global.monitoring_integration_tests.name }}:{{ .Values.global.monitoring_integration_tests.version }}
+        # image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.monitoring_integration_tests.dir }}{{ .Values.global.monitoring_integration_tests.name }}:{{ .Values.global.monitoring_integration_tests.version }}
+        image: eu.gcr.io/sap-se-cx-kyma-wookiee/monitoring-integration-tests:kubernetes-1.16.3
```
Comment (m00g3n) [inline resources/monitoring/templates/tests/test.yaml]: will be updated in image-bump PR

File: resources/service-catalog/values.yaml [hunk 6]
```diff
@@ -1,13 +1,15 @@
 global:
   isLocalEnv: false
   containerRegistry:
-    path: eu.gcr.io/kyma-project
+    # path: eu.gcr.io/kyma-project
+    path: eu.gcr.io/sap-se-cx-kyma-wookiee
```
Comment (m00g3n) [inline resources/service-catalog/values.yaml]: will be updated in image-bump PR

File: resources/xip-patch/values.yaml [hunk 7]
```diff
@@ -1,8 +1,8 @@
 containerRegistry:
-  path: eu.gcr.io/kyma-project
+  path: eu.gcr.io/sap-se-cx-kyma-wookiee
```
Comment (m00g3n) [inline resources/xip-patch/values.yaml]: will be updated in image-bump PR

**Comment by magicmatatjahu** [comment]
@m00g3n Please change version of kubectl and kubernetes (to 1.16.4) and minikube (to 1.16.2) in `installation/scripts/minikube.sh`

File: resources/helm-broker/templates/default-addons-cfg.yaml [hunk 8]
```diff
@@ -8,4 +8,4 @@ metadata:
     addons.kyma-project.io/managed: "true"
 spec:
   repositories:
-  - url: {{ .Values.defautlAddonsURL }}
+  - url: "github.com/kyma-project/addons//addons/index.yaml"
```
Comment (polskikiel) [inline resources/helm-broker/templates/default-addons-cfg.yaml]: We will do addons release 0.11.0 and bump it before the release

File: resources/compass/charts/postgresql/templates/_helpers.tpl [hunk 9]
```diff
@@ -38,7 +38,7 @@ Return the appropriate apiVersion for networkpolicy.
 */}}
 {{- define "postgresql.networkPolicy.apiVersion" -}}
 {{- if semverCompare ">=1.4-0, <1.7-0" .Capabilities.KubeVersion.GitVersion -}}
-"extensions/v1beta1"
```
Comment (pkosiec) [inline resources/compass/charts/postgresql/templates/_helpers.tpl]: Please revert it, it should be as it is

File: components/kyma-operator/Gopkg.toml [hunk 10]
```diff
@@ -60,19 +60,19 @@ required = [
...
 
 [[override]]
   name = "k8s.io/client-go"
-  version = "kubernetes-1.15.3"
+  version = "kubernetes-1.16.3"
 
 [[override]]
   name = "k8s.io/code-generator"
-  version = "kubernetes-1.15.3"
+  version = "kubernetes-1.16.3"
```
Comment (sayanh) [inline components/kyma-operator/Gopkg.toml]: I see you have changed the dependencies to k8s 1.16.* here but not for all components. Is there a reason for that? The functionalities do work though.

**Comment by sayanh** [comment]
test review

File: components/istio-installer/Dockerfile [hunk 11]
```diff
@@ -1,4 +1,4 @@
-FROM eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20190325-ff66a3a
+FROM eu.gcr.io/sap-se-cx-kyma-wookiee/alpine-kubectl:kubernetes-1.16
```
Comment (piotrmsc) [inline components/istio-installer/Dockerfile]: I think we should use official images, created and hosted in kyma-project gcr repo

File: resources/istio/templates/job.yaml [hunk 12]
```diff
@@ -25,7 +25,7 @@ spec:
                 {{- end }}
                 set -e
 {{.Files.Get "files/job-script.sh" | printf "%s" | indent 16}}             
-          image: "eu.gcr.io/kyma-project/istio-installer:308855d7"
+          image: "eu.gcr.io/sap-se-cx-kyma-wookiee/istio-installer:kubernetes-1.16"
```
Comment (piotrmsc) [inline resources/istio/templates/job.yaml]: Same story, it should be official kyma gcr repo

File: components/istio-installer/Dockerfile (hunk 11 above)
Comment (m00g3n) [reply components/istio-installer/Dockerfile]: Yes, I will prepare PRs, I've needed those to set up test cluster.

**Comment by m00g3n** [comment]
@sayanh 
What I actually did is applying minimal set of changes to make Kyma running on Kubernetes 1.16. 
There was no need to migrate all the components - some of them (e.g. event-sources) worked out of the box.

**Pull request merged**